lib/sinon/fake.js

```diff
@@ -5,117 +5,283 @@ var createProxy = require("./proxy");
 var nextTick = require("./util/core/next-tick");
 
 var slice = arrayProto.slice;
+var promiseLib = Promise;
 
-function getError(value) {
-    return value instanceof Error ? value : new Error(value);
-}
-
-var uuid = 0;
-function wrapFunc(f) {
-    var proxy;
-    var fakeInstance = function () {
-        var firstArg, lastArg;
+module.exports = fake;
 
-        if (arguments.length > 0) {
-            firstArg = arguments[0];
-            lastArg = arguments[arguments.length - 1];
-        }
+/**
+ * Returns a `fake` that records all calls, arguments and return values.
+ *
+ * When an `f` argument is supplied, this implementation will be used.
```

Comment:
```suggestion
 * When an `function` argument is supplied, this implementation will be used.
```
is `f` a known shorthand, it might be a barrier to understanding to some